File: Excel/700-799/714/714 Challenge.py
import pandas as pd

path = "700-799/714/714 Sell Alignment.xlsx"

input = pd.read_excel(path, usecols="A:B", skiprows=1, nrows=11)
test = pd.read_excel(path, usecols="D:F", skiprows=1, nrows=3).rename(columns=lambda c: c.replace('.1', ''))

input['Instr'] = input['Produce'].str.endswith('N').map({True: "Don't Sell", False: "Sell"})
input['Produce'] = input['Produce'].str.extract(r'^(\w+)')

result = (
    input.groupby(['Farmer', 'Instr'])['Produce']
    .apply(lambda x: ', '.join(sorted(x.dropna())))
    .unstack()
    .reset_index()[['Farmer', 'Sell', "Don't Sell"]]
    .sort_values('Farmer')
    .reset_index(drop=True)
)
print(result.equals(test)) # True

File: Excel/700-799/714/714 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/714/714 Sell Alignment.xlsx"
input = read_excel(path, range = "A2:B12")
test = read_excel(path, range = "D2:F5")

result = input %>%
  mutate(
    Instr = ifelse(str_ends(Produce, "N"), "Don't Sell", "Sell"),
    Produce = str_extract(Produce, "^[^ ]+")
  ) %>%
  pivot_wider(
    names_from = Instr,
    values_from = Produce,
    values_fn = ~ paste(sort(.), collapse = ", ")
  ) %>%
  arrange(Farmer)

all.equal(result, test)
#> [1] TRUE
